src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterHelper.cs

```diff
@@ -184,22 +186,41 @@ public static void UpdateCoreTypeBlobs(string testDataFilePath, string[] blobs)
                     continue;
                 }
 
+                string pattern = null;
+                string replacement = null;
                 if (PlatformDetection.IsFullFramework)
                 {
-                    testDataLine = Regex.Replace(testDataLine, ", \"AAEAAAD[^\"]+\"(?!,)", ", \"" + blobs[numberOfBlobs] + "\"");
+                    pattern = ", \"AAEAAAD[^\"]+\"(?!,)";
+                    replacement = ", \"" + blobs[numberOfBlobs] + "\"";
                 }
                 else
                 {
-                    testDataLine = Regex.Replace(testDataLine, "\"AAEAAAD[^\"]+\",", "\"" + blobs[numberOfBlobs] + "\",");
+                    pattern = "\"AAEAAAD[^\"]+\",";
+                    replacement = "\"" + blobs[numberOfBlobs] + "\",";
                 }
 
+                Regex regex = new Regex(pattern);
+                Match match = regex.Match(testDataLine);
+                if (match.Success)
+                {
+                    numberOfFoundBlobs++;
+                }
+                string updatedLine = regex.Replace(testDataLine, replacement);
+                if (testDataLine != updatedLine)
+                {
+                    numberOfUpdatedBlobs++;
+                }
+                testDataLine = updatedLine;
+
                 updatedTestDataLines.Add(testDataLine);
                 numberOfBlobs++;
             }
 
             // Check if all blobs were recognized and write updates to file
             Assert.Equal(numberOfBlobs, blobs.Length);
             File.WriteAllLines(testDataFilePath, updatedTestDataLines);
+
+            return (numberOfBlobs, numberOfFoundBlobs, numberOfUpdatedBlobs);
```

Comment:
numberOfUpdatedBlobs is definitely interesting for the caller. numberofFoundBlobs is just a check if the regex pattern is working correctly and numberOfBlobs tells us how many active test cases we currently have.